File: Searching/linearsearch.py
# def linearSearch(arr, N, target):
#     for i in range(0, N):
#         if (arr[i] == target):
#             return i
#     return -1
def linearSearch(arr, N, target,counter):
    if counter < N:
        if (arr[counter] == target):
            return counter
        return linearSearch(arr, N, target, counter+1)
    return -1
arr = [2, 3, 4, 10, 40]
target = 10
N = len(arr)
result = linearSearch(arr, N, target,0)
if(result == -1):
    print("Element is not present in array")
else:
    print("Element is present at index", result)

File: Searching/linearSearch.cpp
#include <iostream>
using namespace std;

// int linearSearch(int arr[], int n, int target) {
//   for(int i = 0; i < n; i++) {
//     if(arr[i] == target) {
//       return i;
//     }
//   }
//   return -1;
// }

int linearSearch(int arr[], int i, int n, int target) {
  if (i < n) {
    if (arr[i] == target) {
      return i;
    }
    return linearSearch(arr, i + 1, n, target);
  }
  return -1;
}

int main() {
  int a[5] = {1, 2, 3, 4, 5};
  int ans = linearSearch(a, 0, 5, 3);
  cout << ans;
  return 0;
}
